### PR TITLE
Upgrade Agora.io SDK

### PR DESCRIPTION
**Pull request by drew-thompson**
# What's in This PR?

A whole bunch of documentation updates, along with the introduction of methods introduced since this library was last published. I've also removed the `@angular/cdk` peer dependency, since I realized it wasn't necessary for the library's operation.

Shoutout to [@spixy](https://github.com/spixy) for the [inspiration](https://github.com/drew-thompson/ngx-agora/pull/13)! I would have merged that in, but it's very difficult to assure no malware was inserted into minified Agora SDK or the `package.lock.json`.

I plan on synchronizing this package version and publishing to NPM in a follow-up PR.

### CODE REVIEW

File: projects/ngx-agora/src/lib/data/types/video-profile.type.ts [hunk 1]
```diff
@@ -1,20 +1,30 @@
...
+ *   In this case, we recommend you use the mainstream video profiles (the ones with the _1 suffix in the above table).
+ * - The Safari browser does not support modifying the video frame rate (30 fps by default).
+ *   If you set a frame rate other than 30 fps on Safari, the browser may change or reject your setting.
+ * - Due to limitations of some devices and browsers, the resolution you set may fail to take effect and get adjusted by the browser.
+ *   In this case, billings are calculated based on the actual resolution.
+ *
  * @see [Table](https://docs.agora.io/en/Video/API%20Reference/web/interfaces/agorartc.stream.html#setvideoprofile) for more details.
  */
 export type VideoProfile =
-  | '120p'
```
Comment (drew-thompson) [inline projects/ngx-agora/src/lib/data/types/video-profile.type.ts:6]: @digitallysavvy why were these resolution settings removed from [the docs](https://docs.agora.io/en/Video/API%20Reference/web/interfaces/agorartc.stream.html#setvideoprofile)?

Comment (digitallysavvy) [reply projects/ngx-agora/src/lib/data/types/video-profile.type.ts:6]: I'm not sure why they were removed, these profiles may no longer exist in v3.0 of The Agora SDK. given 3.0 was a major version its very possible this was a breaking change that was made during the update.

**Comment by drew-thompson** [comment]
> I would have thought the `node_modules` would have been part of the `.gitignore`? 
> 
> 
> 
> Since the npm `package.json` and `package-lock.json` should ensure that the correct versions of each dependency is used, why is it packaged into the library?

That’s a fair point, and probably totally correct. It’s due to the nested `projects/ngx-agora` directory having its own set of dependencies. I can add it to the `.gitignore` and remove it from the repo! I’d like to mention that it’s not a regression, since it currently exists [here](https://github.com/drew-thompson/ngx-agora/tree/master/projects/ngx-agora/node_modules/agora-rtc-sdk).

TLDR; when I made this I didn’t notice that.

File: projects/ngx-agora/src/lib/data/types/video-profile.type.ts (hunk 1 above)
Comment (drew-thompson) [reply projects/ngx-agora/src/lib/data/types/video-profile.type.ts:6]: I'll go by what the docs say, but that definitely should be documented somewhere clear 😆
